exercises/simple-linked-list/example.scala

```diff
@@ -0,0 +1,94 @@
+import scala.reflect.ClassTag
+
+class Node[+T]
+case object Empty extends Node[Nothing]
+case class NonEmpty[T](value: T, var next: Node[T]) extends Node[T]
+
+trait SimpleLinkedList[T] {
+  def isEmpty: Boolean
+  def value: T
+  def add(item: T): SimpleLinkedList[T]
+  def next: SimpleLinkedList[T]
+  def reverse: SimpleLinkedList[T]
+  def toArray[U >: T](implicit m: ClassTag[U]): Array[U]
```

Comment:
Thinking about ClassTag more, if there isn't another solution. This might be something we should comment on in HINTS.md